diff -Naur binutils-2.24.orig/bfd/config.bfd binutils-2.24/bfd/config.bfd
--- binutils-2.24.orig/bfd/config.bfd	2013-11-04 07:33:37.000000000 -0800
+++ binutils-2.24/bfd/config.bfd	2014-08-10 23:15:14.817166328 -0700
@@ -630,6 +630,9 @@
     targ64_selvecs="bfd_elf64_x86_64_nacl_vec bfd_elf32_x86_64_nacl_vec"
     targ_archs="$targ_archs bfd_arm_arch"
     ;;
+  i[3-7]86-*-tacos*)
+    targ_defvec=bfd_elf32_i386_vec
+    ;;
 #ifdef BFD64
   x86_64-*-darwin*)
     targ_defvec=mach_o_x86_64_vec
diff -Naur binutils-2.24.orig/config.sub binutils-2.24/config.sub
--- binutils-2.24.orig/config.sub	2013-11-26 03:37:33.000000000 -0800
+++ binutils-2.24/config.sub	2014-08-10 22:59:25.305896483 -0700
@@ -2,7 +2,7 @@
 # Configuration validation subroutine script.
 #   Copyright 1992-2013 Free Software Foundation, Inc.
 
-timestamp='2013-10-01'
+timestamp='2014-08-10'
 
 # This file is free software; you can redistribute it and/or modify it
 # under the terms of the GNU General Public License as published by
@@ -1357,6 +1357,7 @@
 	      | -sym* | -kopensolaris* | -plan9* \
 	      | -amigaos* | -amigados* | -msdos* | -newsos* | -unicos* | -aof* \
 	      | -aos* | -aros* \
+              | -tacos* \
 	      | -nindy* | -vxsim* | -vxworks* | -ebmon* | -hms* | -mvs* \
 	      | -clix* | -riscos* | -uniplus* | -iris* | -rtu* | -xenix* \
 	      | -hiux* | -386bsd* | -knetbsd* | -mirbsd* | -netbsd* \
diff -Naur binutils-2.24.orig/gas/configure.tgt binutils-2.24/gas/configure.tgt
--- binutils-2.24.orig/gas/configure.tgt	2013-11-04 07:33:37.000000000 -0800
+++ binutils-2.24/gas/configure.tgt	2014-08-10 23:16:10.074329709 -0700
@@ -270,6 +270,7 @@
   i386-*-chaos)				fmt=elf ;;
   i386-*-rdos*)				fmt=elf ;;
   i386-*-darwin*)			fmt=macho ;;
+  i386-*-tacos*)                        fmt=elf ;;
 
   i860-*-*)				fmt=elf endian=little ;;
 
diff -Naur binutils-2.24.orig/ld/configure.tgt binutils-2.24/ld/configure.tgt
--- binutils-2.24.orig/ld/configure.tgt	2013-11-26 03:37:33.000000000 -0800
+++ binutils-2.24/ld/configure.tgt	2014-08-10 23:05:37.613148487 -0700
@@ -234,6 +234,7 @@
 			targ64_extra_libpath=elf_x86_64
 			targ_extra_libpath=elf32_x86_64
 			tdir_i386linux=${targ_alias}aout ;;
+i[3-7]86-*-tacos*)	targ_emul=i386tacos ;;
 x86_64-*-linux-gnux32)	targ_emul=elf32_x86_64
 			targ_extra_emuls="elf_x86_64 elf_i386 i386linux elf_l1om elf_k1om"
 			targ_extra_libpath="elf_i386 elf_x86_64 elf_l1om elf_k1om"
diff -Naur binutils-2.24.orig/ld/emulparams/i386tacos.sh binutils-2.24/ld/emulparams/i386tacos.sh
--- binutils-2.24.orig/ld/emulparams/i386tacos.sh	1969-12-31 16:00:00.000000000 -0800
+++ binutils-2.24/ld/emulparams/i386tacos.sh	2014-08-10 23:25:33.694521061 -0700
@@ -0,0 +1,14 @@
+SCRIPT_NAME=elf
+OUTPUT_FORMAT="elf32-i386"
+NO_RELA_RELOCS=yes
+TEXT_START_ADDR=0x00400000
+MAXPAGESIZE="CONSTANT (MAXPAGESIZE)"
+COMMONPAGESIZE="CONSTANT (COMMONPAGESIZE)"
+ARCH=i386
+MACHINE=
+TEMPLATE_NAME=elf32
+GENERATE_SHLIB_SCRIPT=yes
+GENERATE_PIE_SCRIPT=yes
+NO_SMALL_DATA=yes
+SEPARATE_GOTPLT="SIZEOF (.got.plt) >= 12 ? 12 : 0"
+IREL_IN_PLT=
diff -Naur binutils-2.24.orig/ld/Makefile.in binutils-2.24/ld/Makefile.in
--- binutils-2.24.orig/ld/Makefile.in	2013-11-26 03:37:33.000000000 -0800
+++ binutils-2.24/ld/Makefile.in	2014-08-10 23:26:46.672108449 -0700
@@ -2752,6 +2752,9 @@
 eelf_i386.c: $(srcdir)/emulparams/elf_i386.sh \
   $(ELF_DEPS) $(srcdir)/scripttempl/elf.sc ${GEN_DEPENDS}
 	${GENSCRIPTS} elf_i386 "$(tdir_elf_i386)"
+ei386tacos.c: $(srcdir)/emulparams/i386tacos.sh \
+  $(ELF_DEPS) $(srcdir)/scripttempl/elf.sc ${GEN_DEPENDS}
+	${GENSCRIPTS} i386tacos "$(tdir_i386tacos)"
 eelf_i386_be.c: $(srcdir)/emulparams/elf_i386_be.sh \
   $(ELF_DEPS) $(srcdir)/scripttempl/elf.sc ${GEN_DEPENDS}
 	${GENSCRIPTS} elf_i386_be "$(tdir_elf_i386_be)"
